gpcontrib/arenadata_toolkit/src/tf_shmem.c

```diff
@@ -0,0 +1,215 @@
+#include "bloom_set.h"
+#include "tf_shmem.h"
+
+#include "storage/ipc.h"
+#include "storage/shmem.h"
+
+#include "arenadata_toolkit_guc.h"
+
+#include <math.h>
+
+static shmem_startup_hook_type next_shmem_startup_hook = NULL;
+tf_shared_state_t *tf_shared_state;
+LWLock *tf_state_lock;
+LWLock *bloom_set_lock;
+tf_entry_lock_t bloom_locks[MAX_DB_TRACK_COUNT];
```

Comment:
The bloom_set api allows to init and use several instances of `bloom_set_t`, but they will all use the same locks from `bloom_locks`. That doesn't look right.
I think we should either eliminate the `bloom_set_t *` parameter from the bloom set api (and claim it is always a singleton), or we should use individual lock sets for each bloom_set_t. I'd prefer the second option.

Reply:
Made singleton variant